adal/src/main/java/com/microsoft/aad/adal/BrokerProxy.java

```diff
@@ -892,21 +893,20 @@ private boolean checkAccount(final AccountManager am, String username, String un
                 Account[] accountList = mAcctManager
                         .getAccountsByType(AuthenticationConstants.Broker.BROKER_ACCOUNT_TYPE);
 
+                final BrokerValidator brokerValidator = new BrokerValidator(mContext);
+
                 // For new broker with PRT support, both company portal and
                 // azure authenticator will be able to support multi-user.
-                if (authenticator.packageName
-                        .equalsIgnoreCase(AuthenticationConstants.Broker.AZURE_AUTHENTICATOR_APP_PACKAGE_NAME)
-                        || authenticator.packageName
-                        .equalsIgnoreCase(AuthenticationConstants.Broker.COMPANY_PORTAL_APP_PACKAGE_NAME)
-                        || authenticator.packageName
-                        .equalsIgnoreCase(AuthenticationSettings.INSTANCE.getBrokerPackageName())) {
-                    // Existing broker logic only connects to broker for token
-                    // requests if account exists. New version can allow to
-                    // add accounts through Adal.
-                    if (hasSupportToAddUserThroughBroker(authenticator.packageName)) {
-                        return true;
-                    } else if (accountList.length > 0) {
-                        return verifyAccount(accountList, username, uniqueId);
+                for (final BrokerData brokerData : brokerValidator.getValidBrokers()) {
+                    if (authenticator.packageName.equalsIgnoreCase(brokerData.packageName)) {
```

Comment:
This makes me think we should have a getBrokerData(String packageName) method that returns null or throws an exception if the broker isn't present.

Reply:
Addressed here: https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/1578/commits/22463bc9c354ecd24bc4a36baf62963a13aee513